src/components/account-tab.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import Logger from '../common/logger.js';
+import EmailInput from './email-input.js';
+import { useUser } from './user-context.js';
+import { Form, Button, message } from 'antd';
+import { useTranslation } from 'react-i18next';
+import UsernameInput from './username-input.js';
+import errorHelper from '../ui/error-helper.js';
+import { useService } from './container-context.js';
+import UserApiClient from '../services/user-api-client.js';
+import { SAVE_USER_RESULT } from '../domain/user-management.js';
+import { confirmIdentityWithPassword } from './confirmation-dialogs.js';
+import { formItemLayoutShape, tailFormItemLayoutShape } from '../ui/default-prop-types.js';
+
+const logger = new Logger(import.meta.url);
+
+const FormItem = Form.Item;
+
+function AccountTab({ formItemLayout, tailFormItemLayout }) {
+  const user = useUser();
+  const { t } = useTranslation('accountTab');
+  const userApiClient = useService(UserApiClient);
+
+  const [state, setState] = useState({
+    username: user.username,
+    email: user.email,
+    forbiddenEmails: [],
+    forbiddenUsernames: []
+  });
+
+  const formRef = React.createRef();
+
+  const saveAccountData = async ({ username, email }) => {
+    try {
+      const { result, user: updatedUser } = await userApiClient.saveUserAccount({ username, email });
+
+      switch (result) {
```

Comment:
Maybe we should consider for such cases using the useReducer hook. It would make this testable and a lot easier to read. @adrianaluchian @ahelmberger what do you think?

Reply:
I don't think that it will be easier to read, wouldn't it be basically the same as we have here (just distributed into two places)?

Reply:
Good to keep in mind 🤔 
In this case though I believe it would not make sense as it's a one-trigger event and represents the form save